### PR TITLE
Close the overlay and revert input value when focus is lost

### LINKED ISSUE #27: Overlay should close when focus goes outside the combo-box
Currently, the page can be scrolled using TAB to focus elements outside the overlay which causes the overlay to be displayed in a wrong place.

Simultaneously, the value of the input should revert to the actual selected value.

### PR DESCRIPTION
**Pull request by tehapo**
Fixes #27

[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-combo-box/67)

### CODE REVIEW

File: test/index.html [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
-      'toggling-dropdown.html',
-      'overlay-position.html',
-      'selecting-items.html',
-      'form-input.html',
-      'filtering.html',
-      'keyboard.html',
-      'scrolling.html',
-      'adaptive-overlay.html',
-      'aria.html'
+      // 'vaadin-combo-box-properties.html',
```
Comment (jounik) [inline test/index.html]: You probably already noticed, but this seems like a mistake that these are commented out.

Comment (tehapo) [inline test/index.html]: Yes, these were commented out while trying to get the tests pass in Win10/IE11 in SauceLabs. Tests still fail, which I need to look into tomorrow. After that I'll add all tests back to the test suite. :smile:

**Comment by tehapo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/index.html, line 14 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67#-K4Rk-NBGAZ79cQ-EWWN-r1-14:-K4h27VZ6tPey9g5Ea9h:-499146009) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/6c42566f7358f692c3798f12e7ec2d4ec93588cd/test/index.html#L14)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67)*

**Comment by platosha** [comment]
Reviewed 1 of 4 files at r2, 2 of 2 files at r3.
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/toggling-dropdown.html, line 95 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67#-K4h103XgBl7VKIH7Mn6:-K4h103XgBl7VKIH7Mn7:-1838102227) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/a0bb2a32536fd96173a80b90ce5845a621d9e2e3/test/toggling-dropdown.html#L95)):*
I suggest using `describeIf` in these two tests and switching the conditional predicates to opposite. Like `describeIf(!fullscreen, ...)` instead of `describeSkipIf(fullscreen, ...)`. That would be easier to read and understand.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67)*

**Comment by tehapo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/toggling-dropdown.html, line 95 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67#-K4h103XgBl7VKIH7Mn6:-K4h39UniRVOdNoD8dAh:1360076744) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/a0bb2a32536fd96173a80b90ce5845a621d9e2e3/test/toggling-dropdown.html#L95)):*
Good point, thanks! Fixed.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67)*

**Comment by Saulis** [comment]
Reviewed 1 of 2 files at r4, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-combo-box.html, line 317 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67#-K4giPA5HDKScjH7O9S1:-K4giPA5HDKScjH7O9S2:-1172479120) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/a0bb2a32536fd96173a80b90ce5845a621d9e2e3/vaadin-combo-box.html#L317)):*
Is `this.async` really needed?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/67)*

**Pull request merged**